CIP-0100/README.md

```diff
@@ -0,0 +1,298 @@
+ - Tool authors MAY choose which contexts to support, but MUST make a best effort to display the metadata in the presence of unrecognized context, up to and including gracefully falling back to a raw display of the JSON document.
+
+**Extensions to the governance metadata standard can take one of two forms**:
+- CIP-100 itself can be updated through the normal CIP process to provide additional clarity on any concepts that are giving people trouble with adoption, or to correct inaccuracies.
+- A new CIP, defining a new vocabulary for a specific use case.
+
+### Governance Transaction Types
+
+In CIP-1694 (and likely any alternative or future evolution of it), there are a number of certificates that can be attached to a transaction pertaining to governance; each of these is equipped with an "anchor", which can refer to an external document. This CIP provides the following JSON-LD schema intended to represent a fairly minimal vocabulary :
+
+ - [Common Definitions](./cip-100.common.jsonld)
+
+Additionally, someone may wish to augment a previous piece of metadata with new information, divorced from the transaction that initially published it; this may be useful, for example, to provide additional arguments in favor of or against a proposal, provide translations to other languages, provide a layman's explanation of a highly technical document, etc.
+
+These can, in theory, be published anywhere, but to aide in discoverability for indexing tools, we explicitly call out the Cardano blockchain as a convenient "public square" in which to publish this metadata.
+
+To this end, this CIP reserves [metadatum label 1694](../CIP-0010/README.md) for publishing these kinds of augments on-chain. This CIP also provides a minimal JSON-LD schema specifically for these documents, though the vocabulary may be extended in the same way that any other json-ld context can be extended:
+
+ - [Augmentation](./cip-100.augmentation.jsonld)
+
+The rest of this document will provide a high level description of the properties described in each of the schemas above.
+
+#### Common properties
+
+The following properties are considered common to all types of transactions, and the minimal set needed for "minimum viable governance":
+
+- hash-algorithm: The algorithm used to hash the document for the purposes of the on-chain anchor; currently only supports blake2b-256
+- authors: The primary contributors and authors for this metadata document
+  - An array of objects
+  - Each object MAY have a display name
+  - Each object MUST have one or more witnesses
+  - Each witness may define a context, describing the manner in which the document has been witnessed
+  - A default witness scheme is described in a later section
+  - Tooling authors SHOULD validate these witnesses; if absent or invalid, tooling authors SHOULD make this clear in the UI
+- body: The material contents of the document, separate for the purposes of signing
+  - justification
```

Comment:
```suggestion
```
Id argue that a generic "justification" is not needed here, and such longer form fields should be left to subsequent standards.
As "justification" could easily be broken down into smaller more specific sections depending on the type of anchor, be it governance action OR DRep registration.

Reply:
I think some form of free-form text input should be considered part of the minimal definition; It's an optional field, so leaving it off does no harm, but without it, there's very little point in attaching any metadata at all.

Another CIP can (and should) extend, replace, or supplement the justification field, but I find it hard to see how it could cause issues. It's just a free form text field for the user.

Reply:
I could definitely be convinced to give it a different name, though, if you have a suggestion.

Reply:
This thought came when I was writing https://github.com/cardano-foundation/CIPs/pull/632, and I wasn't sure how the proposed motivation and rationale should relate to justification.

Im not sure how is best to relate these fields, perhaps a rename of justification will help?

Reply:
Perhaps just "Comment"?

Reply:
Ultimately the reason for the extension mechanism is to enable diversity of fields, though, so personally I still don't see a problem with "Justification", "Rational", *and* "Motivation" fields, even if the definition overlaps a little bit; It is perhaps a bit of an "aesthetic violation", but there's no material harm / downside in my eyes.

Reply:
> Perhaps just "Comment"?

Agreed, I think this works better.

Reply:
(It's also worth noting, though, that this CIP explicitly supports reuse of terms; the point of JSON-LD is to fully namespace every property, so that you don't have to worry about conflicting. It's probably best not to start off immediately with a clash though lol)